sdk/keyvault/Azure.Security.KeyVault.Keys/perf/Scenarios/Sign.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Security.Cryptography;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Security.KeyVault.Keys.Cryptography;
+using Azure.Security.KeyVault.Keys.Perf.Infrastructure;
+using Azure.Test.Perf;
+using NUnit.Framework;
+
+namespace Azure.Security.KeyVault.Keys.Perf.Scenarios
+{
+    public sealed class Sign : CryptographyScenarioBase<PerfOptions>
+    {
+        private static readonly SignatureAlgorithm s_algorithm = SignatureAlgorithm.RS256;
+
+        private readonly SHA256 _hasher;
+        private byte[] _digest;
+
+        public Sign(PerfOptions options) : base(options)
+        {
+            _hasher = SHA256.Create();
+        }
+
+        public override async Task GlobalSetupAsync()
+        {
+            await base.GlobalSetupAsync();
+
+            // CryptographyClient caches the public key so signing once removes the initial request from metrics.
```

Comment:
The perf framework already has a `warmup` phase separate from the `duration` test run.  So you shouldn't need to manually call `SignAsync()` or `Regenerate()` in `GlobalSetup/Setup`.

Reply:
That's a good point. I admit I completed missed just taking advantage of that here.

Per offline discussion, the docs should probably be updated to mention that Setup(Async) is called once before Run(Async), but that Run(Async) can be called numerous times after that per instance of the class (if I reiterated that right).